### DISCUSSION TITLE
Anyone using animated transitions between website pages?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#925] [Ask a question] [answered]
Hello all!
So far, I have not used animated transitions between website pages.
Have any of you ever tested this together with Yellow? Maybe [barba.js](https://barba.js.org/), [swup.js](https://swup.js.org/) or similar solutions? Can you recommend a solution?

**Edit:**
I like the page from [Pandolfini](https://www.pandolfini.com.au/). All jumps between pages are very smooth there.

**Comment by schulle4u** [comment, marked as answer]
Even if I can relate to this idea for e.g. app-like websites, I never had the thought to implement something like this for Yellow. However this is mostly due to the fact that I'm a keyboard/screen reader user, and most JS transitions I've seen so far seem to have a slower performance than loading the page directly. But this is just my experience, I've never proven it with facts. 🙂

**Comment by PetersOtto** [comment]
I think it always depends on the purpose of the website. A website for a rock band or a movie looks very different from a wiki page. A wiki page certainly doesn't need this effect. In general, I find the effect visually appealing. I think I'll give it a try. Let's see how difficult it is to integrate it into Yellow.

**Comment by PetersOtto** [comment]
I have tested [barba.js](https://barba.js.org/) in combination with [anime.js](https://animejs.com/). I like it 🙂
Here is a quick and dirty example. Maybe someone would like to test it.

**1. header.html**
```
<script src="https://cdn.jsdelivr.net/npm/@barba/core"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/animejs/3.2.2/anime.js" integrity="sha512-FSNQjUUq97PtDnAJyMlBTH8EhV4o5X5FLVbzS9ttrdEVVbmrssJBh/PgXFQbxS+6yYx5n5irOj7de6Th0TRQZA==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```
and change this ...
`<body>`
to
`<body data-barba="wrapper">`

and change this ...
`<div class="<?php echo "page layout-".$this->yellow->page->getHtml("layout") ?>" >`
to
`<div class="<?php echo "page layout-".$this->yellow->page->getHtml("layout") ?>" data-barba="container" data-barba-namespace="home">`


**2. footer.html**
```
<script>
    barba.init({
    transitions: [{
        async leave() {
            await anime({
                duration: 250,
                targets: '.main',
                opacity: ['1', '0'],
                easing: 'easeOutSine',
            }).finished;
        },
        enter() {
            anime({
                duration: 250,
                targets: '.main',
                opacity: ['0', '1'],
                easing: 'easeInSine',
            });
        }
    }]
});
</script>
```
Good Luck!
